src/extension/extensionServer.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as em from "../common/extensionMessaging";
+import * as http from "http";
+import {ILaunchArgs} from "../common/launchArgs";
+import {Packager} from "../common/packager";
+import * as Q from "q";
+import {SettingsHelper} from "./settingsHelper";
+import * as vscode from "vscode";
+
+
+export class ExtensionServer implements vscode.Disposable {
+
+    private serverInstance: http.Server = null;
+    private messageHandlerDictionary: { [id: number]: ((args?: any[]) => Q.Promise<any>) } = {};
+    private reactNativePackager: Packager;
+
+    public constructor(reactNativePackager: Packager) {
+        this.reactNativePackager = reactNativePackager;
+
+        /* register handlers for all messages */
+        this.messageHandlerDictionary[em.ExtensionMessage.START_PACKAGER] = this.startPackager;
+        this.messageHandlerDictionary[em.ExtensionMessage.STOP_PACKAGER] = this.stopPackager;
+        this.messageHandlerDictionary[em.ExtensionMessage.PREWARM_BUNDLE_CACHE] = this.prewarmBundleCache;
+    }
+
+    /**
+     * Starts the server.
+     */
+    public setup(): Q.Promise<void> {
+
+        let deferred = Q.defer<void>();
+
+        let requestHandler = (request: http.IncomingMessage, response: http.ServerResponse) => {
+            this.handleIncomingMessage(request, response);
+        };
+
+        let launchCallback = (error: any) => {
+            if (error) {
+                deferred.reject(error);
+            } else {
+                deferred.resolve(null);
+            }
+        };
+
+        let port = em.ServerDefaultParams.PORT;
+        SettingsHelper.readLaunchJson()
```

Comment:
I believe that we also need a file watcher or to hook onto some vscode event if there is one for launch.json updating. If the user updates the launch.json we should restart the server respecting the new value